### PR TITLE
[ENG-5109] Add tests for Addon Provider Class

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-5109]
-   Feature flag: n/a

## Purpose
- Add test for new Addon Provider class

## Summary of Changes
- Add the test
- Update some logic in the Provider class
- Update some models and mirage stuff

## Screenshot(s)
- NA

## QA Notes

[ENG-5109]: https://openscience.atlassian.net/browse/ENG-5109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mirage/serializers/internal-resource.ts [hunk 1]
```diff
@@ -11,7 +10,7 @@ export default class InternalResourceSerializer extends ApplicationSerializer<In
             configuredStorageAddons: {
                 links: {
                     related: {
-                        href: `${addonServiceAPIUrl}internal_resources/${model.id}/configured_storage_addons`,
+                        href: `${addonServiceAPIUrl}internal-resources/${model.id}/configured-storage-addons`,
```
Comment (aaxelb) [inline mirage/serializers/internal-resource.ts:13]:
```suggestion
                        href: `${addonServiceAPIUrl}internal-resources/${model.id}/configured-storage-addons/`,
```
to match https://github.com/CenterForOpenScience/ember-osf-web/pull/2107/files#diff-34e59843696a4ce1ffea7d2707d97d08851e7ab33b4958854d6412e56b47ecd3R538 ?

File: tests/unit/packages/addons-service/provider-test.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            authorizedResource: internalResource,
+        });
+
+        const provider = new Provider(externalStorageService, currentUser, node);
+
+        assert.notOk(provider.internalUser, 'Provider internalUser is not set before initialize');
+        assert.notOk(provider.serviceNode, 'Provider serviceNode is not set before initialize');
+        assert.notOk(provider.configuredStorageAddon, 'Provider configuredStorageAddon is not set before initialize');
+
+        await settled();
```
Comment (brianjgeiger) [inline tests/unit/packages/addons-service/provider-test.ts:56]: Proooobably won't have any timing issues related with doing it this way. I wonder if they ever made changes to ember if that would change, though.

File: app/models/configured-storage-addon.ts [hunk 3]
```diff
@@ -11,10 +11,10 @@ export default class ConfiguredStorageAddonModel extends OsfModel {
     @attr('fixstring') externalUserDisplayName!: string;
     @attr('fixstring') rootFolder!: string;
 
-    @belongsTo('external-storage-service')
+    @belongsTo('external-storage-service', { inverse: null })
     storageProvider!: AsyncBelongsTo<ExternalStorageServiceModel> & ExternalStorageServiceModel;
 
-    @belongsTo('internal-user', { inverse: 'configuredResources' })
+    @belongsTo('internal-user')
```
[the remark this answers is not in the document]
Comment (futa-ikeda) [reply app/models/configured-storage-addon.ts:17]: Good catch!

File: mirage/scenarios/dashboard.ts [hunk 4]
```diff
@@ -108,13 +108,14 @@ export function dashboardScenario(server: Server, currentUser: ModelInstance<Use
     const boxAccount = server.create('authorized-storage-account', {
         externalUserId: currentUser.id,
         externalUserDisplayName: currentUser.fullName,
-        defaultRootFolder: '/',
+        defaultRootFolder: '/root/',
```
[the remark this answers is not in the document]
Comment (futa-ikeda) [reply mirage/scenarios/dashboard.ts:111]: Gotcha. I recall some other Addons API-related updates that will be needed (I think around storage-provider capabilities), so will ask in slack

File: tests/unit/packages/addons-service/provider-test.ts (hunk 2 above)
Comment (futa-ikeda) [reply tests/unit/packages/addons-service/provider-test.ts:56]: Yea, I'm not sure if these `notOk` asserts are really needed and they may eventually cause some weird breakage down the line if the test's async behavior changes. I think I will get rid of the `assert.notOk`s, as I don't think it's adding much value

**Pull request merged**